main.go

```diff
@@ -272,3 +272,22 @@ func startPollServer(out chan<- *net.IP, localIp *net.IP) {
 		}
 	}()
 }
+
+func readSecret(envName string) string {
+	password := os.Getenv(envName)
+
+	if password != "" {
+		return password
+	}
+
+	passwordFilePath := os.Getenv(envName + "_FILE")
+	if passwordFilePath != "" {
+		content, err := os.ReadFile(passwordFilePath)
+		if err != nil {
+			slog.Error("Failed to read DynDns server password from file", logging.ErrorAttr(err))
```

Comment:
And this message should be more generic as well logging that it's unable to read the file specified in $envName.